src/Adapter/Media/MediaServerConfiguration.php

```diff
@@ -61,27 +61,23 @@ public function getConfiguration()
      */
     public function updateConfiguration(array $configuration)
     {
-        $errors = [];
-        $isValid = $this->validateConfiguration($configuration);
-        if (true === $isValid) {
-            $serverOne = $configuration['media_server_one'];
-            $serverTwo = $configuration['media_server_two'];
-            $serverThree = $configuration['media_server_three'];
-
-            $this->configuration->set('PS_MEDIA_SERVER_1', $serverOne);
-            $this->configuration->set('PS_MEDIA_SERVER_2', $serverTwo);
-            $this->configuration->set('PS_MEDIA_SERVER_3', $serverThree);
-
-            if (!empty($serverOne) || !empty($serverTwo) || !empty($serverThree)) {
-                $this->configuration->set('PS_MEDIA_SERVERS', 1);
-            } else {
-                $this->configuration->set('PS_MEDIA_SERVERS', 0);
-            }
-        } else {
-            $errors = $isValid;
+        $validateResult = $this->validateConfiguration($configuration);
+        if ($validateResult !== true) {
+            return $validateResult;
         }
 
-        return $errors;
+        $serverOne = $configuration['media_server_one'];
+        $serverTwo = $configuration['media_server_two'];
+        $serverThree = $configuration['media_server_three'];
+
+        $this->configuration->set('PS_MEDIA_SERVER_1', $serverOne);
+        $this->configuration->set('PS_MEDIA_SERVER_2', $serverTwo);
+        $this->configuration->set('PS_MEDIA_SERVER_3', $serverThree);
```

Comment:
@atomiix Could you tell me how I can do it? I have searched the prestashop source code and of the more than 100 uses of configuration, 0 use the 3rd parameter $shopConstraint.
and I see that all the ones I have found use `ConfigurationInterface` instead of `ShopConfigurationInterface`

Reply:
@davidglezz You can find some informations in these issue : #25058 (and check all connected PR)

Reply:
I think it's ok not to do it in this PR...

It would mean extending `AbstractMultistoreConfiguration` and in the method `updateCondiguration` you would have to get the shop constraint, like this:

`$shopConstraint = $this->getShopConstraint();`

Then to update the configuration values with the new `updateConfigurationValue` method:

`$this->updateConfigurationValue(...)`

But you would also have to add a new `buildResolver` method (which is mandatory to respect the contract of the parent abstract class)

We usually do all of this when we migrate a form in multistore, but for now, it seems overkill and out of the scope of this PR...   

If you want to do this though, you can look at `src/Adapter/Shop/MaintenanceConfiguration.php`

Reply:
@davidglezz so you can tell us, if you want to do it we can help, otherwise your PR can be accepted as is 👍

Reply:
Done